autocertkit/test_generators.py

```diff
@@ -284,6 +283,77 @@ def get_device_config(self):
         rec = utils.combine_recs(rec, utils.get_system_info())
         return rec
 
+##############################################################################
+
+class DeviceXMLGenerator(object):
+
+    TAGS = []
+    CLS = None
+
+    def __init__(self, session, config, mode, network_ifs, storage_devs):
+        self.session = session
+        self.config = config
+        self.mode = mode
+        self.network_ifs = network_ifs
+        self.storage_devs = storage_devs
+
+    def should_generate(self):
+        return self.mode in self.TAGS or self.mode == 'ALL'
+
+    def append_xml_config(self, doc, devices_node):
+        if self.should_generate():
+            self._append_xml_config(doc, devices_node)
+
+    def _append_xml_config(self, doc, devices_node):
+        gen = self.CLS(self.session, self.config)
+        gen.append_xml_config(doc, devices_node)
+
+class NetworkAdaptersXMLGenerator(DeviceXMLGenerator):
+
+    TAGS = ["NET"]
+
+    def _append_xml_config(self, doc, devices_node):
+        for iface in self.network_ifs:
+            natg = NetworkAdapterTestGenerator(self.session, self.config, iface)
+            natg.append_xml_config(doc, devices_node)
+
+
+class StorageAdaptersXMLGenerator(DeviceXMLGenerator):
+
+    TAGS = ["LSTOR"]
+
+    def _append_xml_config(self, doc, devices_node):
+        for dev in self.storage_devs:
+            lstg = StorageTestGenerator(self.session, self.config, dev)
+            lstg.append_xml_config(doc, devices_node)
+
+
+class ProcessorsXMLGenerator(DeviceXMLGenerator):
+
+    TAGS = ["CPU"]
+    CLS = ProcessorTestGenerator
+
+class OperationsXMLGenerator(DeviceXMLGenerator):
+
+    TAGS = ["OPS"]
+    CLS = OperationsTestGenerator
+
+
+XML_GENERATORS = [
+    NetworkAdaptersXMLGenerator,
+    StorageAdaptersXMLGenerator,
+    ProcessorsXMLGenerator,
+    OperationsXMLGenerator,
+]
+
+# Support the loading of additional tests
```

Comment:
Can we use try for this instead of using isfile?
try:
    import addons
    XML_GENERATORS.extend(addons.XML_GENERATORS)
except:
    log.debug("No addon found...")

something like above...